### PR TITLE
[improvement](jdbc catalog) Optimize connection pool parameter settings

### PR DESCRIPTION
**Pull request by zy-kkk**
## Proposed changes

Issue Number: close #xxx

This PR makes the following changes to the connection pool of JDBC Catalog
1. Set the maximum connection survival time, the default is 30 minutes

-   Moreover, one-half of the maximum survival time is the recyclable time,
-   One-tenth is the check interval for recycling connections

2. Keepalive only takes effect on the connection pool on BE, and will be activated based on one-fifth of the maximum survival time.
3. The maximum number of existing connections is changed from 100 to 10
4. Add the connection cache recycling thread on BE, and add a parameter to control the recycling time, the default is 28800 (8 hours)
5. Add CatalogID to the key of the connection pool cache to achieve better isolation, requires refresh catalog to take effect
6. Upgrade druid connection pool to version 1.2.20
7. Added JdbcResource's setting of default parameters when upgrading the FE version to avoid errors due to unset parameters.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zy-kkk** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/jdbc/JdbcExternalCatalog.java [hunk 1]
```diff
@@ -236,10 +235,34 @@ public void setDefaultPropsWhenCreating(boolean isReplay) throws DdlException {
...
+            throw new DdlException(
+                    "connection_pool_max_size must be greater than or equal to connection_pool_min_size");
+        }
+        if (connectionPoolMaxWaitTime < 0) {
+            throw new DdlException("connection_pool_max_wait_time must be greater than or equal to 0");
+        }
+        if (connectionPoolMaxWaitTime > 30000) {
+            throw new DdlException("connection_pool_max_wait_time must be less than or equal to 30000");
+        }
+        if (connectionPoolMaxLifeTime < 150000) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/jdbc/JdbcExternalCatalog.java:258]: We need to define this value somewhere, there are too many magic number in code

**Comment by zy-kkk** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — COMMENTED** [review]
Please add some ut or test case to test these parameters

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by BePPPower — APPROVED** [review]

**Pull request merged**